core/trino-main/src/main/java/io/trino/metadata/MetadataManager.java

```diff
@@ -15,11 +15,18 @@
 
 import com.google.common.annotations.VisibleForTesting;
```

Comment:
1. Please add explanation in commit message why do you cache resolved function and what was the biggest performance issue.
2. Please make benchmarks results part of commit message